akvo/rsr/models/organisation.py

```diff
@@ -278,13 +277,7 @@ def knowledge(self):
         def all_projects(self):
             "returns a queryset with all projects that has self as any kind of partner"
             from .project import Project
-            return Project.objects.filter(partnerships__organisation__in=self).distinct()
-
-        def public_projects(self):
-            """
-            Returns a queryset with all public projects that has self as any kind of partner.
-            """
-            return self.all_projects().public()
+            return Project.objects.of_partner(self).distinct()
```

Comment:
We have a function `of_partners` and `of_partner` on `Project.QuerySet`.  `of_partners` works with a list/queryset of organisations, while `of_partner` returns all the projects of a single organisation. Methods on the `QuerySet` essentially need to return a `queryset` of the same model, while methods on the model itself can do literally anything.  I don't exactly understand what the question is, but I don't see a problem with having this method.